cli/tools/task.rs

```diff
@@ -105,24 +106,85 @@ pub async fn execute_script(
         .unwrap()
         .to_owned(),
     };
-    let script = get_script_with_args(script, cli_options);
-    output_task(task_name, &script);
-    let seq_list = deno_task_shell::parser::parse(&script)
-      .with_context(|| format!("Error parsing script '{task_name}'."))?;
-    let npx_commands = resolve_npm_commands(npm_resolver, node_resolver)?;
-    let env_vars = collect_env_vars();
-    let local = LocalSet::new();
-    let future =
-      deno_task_shell::execute(seq_list, env_vars, &cwd, npx_commands);
-    let exit_code = local.run_until(future).await;
-    Ok(exit_code)
+
+    let mut pre_task_name: String = "pre".to_owned();
+    pre_task_name.push_str(task_name);
+    if let Some(pre_script) = package_json_scripts.get(&pre_task_name) {
+      let exit_code = run_npm_script(
+        &pre_task_name,
+        pre_script,
+        cli_options,
+        &cwd,
+        npm_resolver,
+        node_resolver,
+      )
+      .await?;
+
+      if exit_code > 0 {
+        return Ok(exit_code);
+      }
+    }
+
+    let exit_code = run_npm_script(
+      task_name,
+      script,
+      cli_options,
+      &cwd,
+      npm_resolver,
+      node_resolver,
+    )
+    .await?;
+
+    if exit_code > 0 {
+      return Ok(exit_code);
+    }
+
+    let mut post_task_name: String = "post".to_owned();
+    post_task_name.push_str(task_name);
+    if let Some(post_script) = package_json_scripts.get(&post_task_name) {
+      let exit_code = run_npm_script(
+        &post_task_name,
+        post_script,
+        cli_options,
+        &cwd,
+        npm_resolver,
+        node_resolver,
+      )
+      .await?;
+
+      if exit_code > 0 {
+        return Ok(exit_code);
+      }
```

Comment:
These three code blocks are very similar. Perhaps instead we could build up a vector of task_names then iterate through them?

```rs
let task_names = vec![format!("pre{}", task_name"), task_name.clone(), format!("post{}", task_name)];
for task_name in task_names {
  if let Some(script) = package_json_scripts.get(&task_name) {
      // ...
  }
}
```

Reply:
Shouldn't we try to run `pre-` and `post-` task only if the requested task actually is defined?

Reply:
That's what `if let Some(script) = package_json_scripts.get(&task_name) {` is for :)

Reply:
This is great feedback! I wasn't sure if rust folks prefer to use the `format!` macro for string concatenation or not.

Reply:
@dsherret Alright, updated the code with your suggestions. They make the code much more concise compared to what I had before. Thanks again for the feedback 👍